tmt/steps/prepare/__init__.py

```diff
@@ -223,7 +224,8 @@ def as_key(self) -> frozenset['tmt.base.DependencySimple']:
                     continue
 
                 collected_requires[guest].dependencies += tmt.base.assert_simple_dependencies(
-                    phase.essential_requires(),
+                    phase.essential_requires(),  # type: ignore[attr-defined]
+                    # mypy thinks that phase is Phase type, while its actually PluginClass
```

Comment:
Nitpicking: can you put the comment *before* the line with the waiver, and include the waiver? Not often, but it sometimes happen there are more issues to be waived, and then it's nice to have a comment for each of them:

```python
# ignore[attr-defined]: mypy thinks that phase is Phase type, while its actually PluginClass
phase.essential_requires(),  # type: ignore[attr-defined]
```

Reply:
Addressed in https://github.com/teemtee/tmt/pull/3142/commits/feeabd484f3a97e109a126f465d56cebddda694c

Reply:
```suggestion
                    # ignore[attr-defined]: mypy thinks that phase is Phase type, while its actually PluginClass
```